app/src/main/java/com/xmartlabs/doapp/ui/TaskAdapter.java

```diff
@@ -0,0 +1,123 @@
+package com.xmartlabs.doapp.ui;
+
+import android.support.annotation.MainThread;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.xmartlabs.doapp.model.Task;
+import com.xmartlabs.template.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import rx.functions.Action1;
+
+/**
+ * Created by scasas on 3/15/17.
+ */
+public class TaskAdapter extends RecyclerView.Adapter<TaskAdapter.TaskHolder> {
+  private List<Task> tasks = new ArrayList<>();
+  private final Action1<Task> onTaskTapped;
+
+  @MainThread
+  public void setTasks(List<Task> tasks) {
```

Comment:
I don't think that, because the the first time the database will be empty so when the app try to retrieve the data from the task table it would return null.